src/Infrastructure.Styles.Analyzer/RMSYN1104OperatorChainingWithNewlineAnalyzer.cs

```diff
@@ -0,0 +1,137 @@
+
+    internal const string Category = "Readability";
+
+    private static readonly DiagnosticDescriptor Rule = new DiagnosticDescriptor(
+        DiagnosticId,
+        Title,
+        Message,
+        Category,
+        DiagnosticSeverity.Warning,
+        true,
+        Description);
+
+    public override void Initialize (AnalysisContext context)
+    {
+      context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
+
+      context.EnableConcurrentExecution();
+
+      context.RegisterSyntaxNodeAction(AnalyzeBinaryExpressionWithToken,
+          SyntaxKind.AddExpression,
+          SyntaxKind.SubtractExpression,
+          SyntaxKind.MultiplyExpression,
+          SyntaxKind.DivideExpression,
+          SyntaxKind.ModuloExpression);
+
+      context.RegisterSyntaxNodeAction(AnalyzeBinaryExpressionWithToken,
+          SyntaxKind.LessThanExpression,
+          SyntaxKind.LessThanOrEqualExpression,
+          SyntaxKind.GreaterThanOrEqualExpression,
+          SyntaxKind.GreaterThanExpression,
+          SyntaxKind.EqualsExpression,
+          SyntaxKind.NotEqualsExpression);
+
+      context.RegisterSyntaxNodeAction(AnalyzeBinaryExpressionWithToken,
+          SyntaxKind.LogicalOrExpression,
+          SyntaxKind.LogicalAndExpression,
+          SyntaxKind.BitwiseOrExpression,
+          SyntaxKind.BitwiseAndExpression,
+          SyntaxKind.ExclusiveOrExpression);
+
+      context.RegisterSyntaxNodeAction(AnalyzeBinaryExpressionWithToken,
+          SyntaxKind.IsExpression);
+
+      context.RegisterSyntaxNodeAction(AnalyzeNonBinaryExpressionWithTokens,
+          SyntaxKind.IsPatternExpression,
+          SyntaxKind.OrPattern,
+          SyntaxKind.AndPattern);
+
+      // ? part of ?. operator
+      context.RegisterSyntaxNodeAction(AnalyzeConditionalAccessExpression, SyntaxKind.ConditionalAccessExpression);
+      // . part of ?. operator
+      context.RegisterSyntaxNodeAction(AnalyzeNonBinaryExpressionWithTokens, SyntaxKind.MemberBindingExpression);
+
+      // ??
+      context.RegisterSyntaxNodeAction(AnalyzeBinaryExpressionWithToken, SyntaxKind.CoalesceExpression);
+
+      // .ToString().Whatever().UndSoWeiter
+      context.RegisterSyntaxNodeAction(AnalyzeMemberAccessExpression, SyntaxKind.SimpleMemberAccessExpression);
+
+      //ternary operator, aka: something ? yes : no
+      context.RegisterSyntaxNodeAction(AnalyzeNonBinaryExpressionWithTokens, SyntaxKind.ConditionalExpression);
+    }
+
+    public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics { get; } = ImmutableArray.Create(Rule);
+
+    // Fürs review: hier haben wir 3 mal basically die gleiche Methode, das kann man glaube ich irgendwie reduzieren, 
+    // ich weiß allerdings nicht wie. Pls advise me :)
+    // Nur infomäßig, BinaryExpression, MemberAcessExpression etc. erben alle von expression, die property wird allerdings
+    // lokal deklariert
```

Comment:
I think the way it is now is fine as there is no a lot of extra code necessary. To "simplify" it is possible to use generics and a lambda as parameter that does the expression specific selection of the token. This would remove the `ReportDiagnosticIfContainsTrailingEndLine(..., context);` duplication.

As I have said I am fine with this solution so its up to you if you want to "simplify" further

Reply:
Dev jour fixe time :)